### PR TITLE
Update rubocop

### PR DESCRIPTION
**Pull request by mvz**
## Summary

Update RuboCop to 0.50.0

## Details

* Update RuboCop
* Move TODO items to `.rubocop_todo.yml` to allow auto-generation.
* Run RuboCop in a separate Travis stage.

## Motivation and Context

Follow-up of #511.

## Types of changes

- Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

**Comment by olleolleolle** [comment]
I wonder, did you ever see [`mry`](https://github.com/pocke/mry), the tool to rename `.rubocop.yml` rules as they move on? It's pretty rad.

**Comment by mvz** [comment]
> did you ever see `mry`

No, but it looks really useful.

**Comment by mvz** [comment]
Unfortunately, RuboCop 0.51 requires Ruby >= 2.1, so we can either:
* Keep using the old version, or
* Not run RuboCop on older platforms.

**Comment by olleolleolle** [comment]
Can you use `TargetRubyVersion: 1.9` in the .rubocop.yml and then evict the running of RuboCop to a single 2.4.1 "Travis Build Stage", to be run before the tests (their own Test stage).

(Does 0.51.0 even support `TargetRubyVersion: 1.9` any longer?)

**Comment by mvz** [comment]
It turns out `0.50.0` supports that target version, so let's see how things go with that version.

**Comment by olleolleolle** [comment]
@mvz Thank you for going the distance with this issue.

**Comment by olleolleolle** [comment]
@mvz To back up my "why don't you just" about RuboCop: 

Here's a Travis configuration, which uses the built-in, implicit 'test' Stage, and adds a 'codestyle' stage.

https://github.com/egonSchiele/contracts.ruby/blob/master/.travis.yml

**Comment by mvz** [comment]
Right, so rubocop 0.50 also doesn't install on 1.9.3. @olleolleolle I'll go with your suggestion.

**Comment by olleolleolle** [comment]
@mvz I might be that you must decide on `script` yourself, somehow.

**Comment by mvz** [comment]
I'll actually read the docs, now ;-)

**Comment by maxmeyer** [comment]
@mvz Is this still work in progress?

**Comment by mvz** [comment]
@maxmeyer yes, I need to get the separate Travis build for RuboCop working.

File: Rakefile [hunk 1]
```diff
@@ -6,10 +6,13 @@ require 'aruba/platform'
 require 'bundler'
 Bundler.setup
 
-task :default => :test
+task default: [:lint, :test]
```
Comment (olleolleolle) [inline Rakefile]: Do we need both here?

File: Rakefile [hunk 2]
```diff
@@ -6,10 +6,13 @@ require 'aruba/platform'
...
 
-desc 'Run the whole test suite. Any failure will stop rake going on'
-task :test => %w(lint:travis lint:coding_guidelines lint:licenses test:rspec test:cucumber test:cucumber_wip)
+desc 'Run all linters.'
+task test: %w(lint:travis lint:coding_guidelines lint:licenses)
```
Comment (olleolleolle) [inline Rakefile]: Ah, sorry: this is labeled wrong: `task lint: ...` it should be!

**Comment by olleolleolle** [comment]
@mvz I followed some error messages, and found a typo.

**Comment by mvz** [comment]
The Travis build was fixed.

**Pull request merged**